static/app/components/search/index.tsx

```diff
@@ -102,166 +104,147 @@ type Props = WithRouterProps<{orgId: string}> & {
+
+        debouncedSaveQueryMetrics(searchQuery);
+
+        const renderItem =
+          typeof props.renderItem === 'function'
+            ? props.renderItem
+            : DefaultSearchProps.renderItem;
+
+        return (
+          <SearchWrapper>
+            {props.renderInput({getInputProps})}
+
+            {isValidSearch && isOpen ? (
+              <SearchSources
+                searchOptions={props.searchOptions}
+                query={searchQuery}
+                params={props.params}
+                sources={props.sources ?? DefaultSearchProps.sources}
+              >
+                {({isLoading, results, hasAnyResults}) => (
+                  <DropdownBox className={props.dropdownStyle}>
+                    {isLoading ? (
+                      <LoadingWrapper>
+                        <LoadingIndicator mini hideMessage relative />
+                      </LoadingWrapper>
+                    ) : !hasAnyResults ? (
+                      <EmptyItem>{t('No results found')}</EmptyItem>
+                    ) : (
+                      results.slice(0, props.maxResults).map((resultObj, index) => {
+                        return React.cloneElement(
+                          renderItem({
                             index,
-                            highlightedIndex,
-                            getItemProps,
-                          })
-                        )}
-                      {!isLoading && !hasAnyResults && (
-                        <EmptyItem>{t('No results found')}</EmptyItem>
-                      )}
-                      {!isLoading && resultFooter && (
-                        <ResultFooter>{resultFooter}</ResultFooter>
-                      )}
-                    </DropdownBox>
-                  )}
-                </SearchSources>
-              ) : null}
-            </SearchWrapper>
-          );
-        }}
-      </AutoComplete>
-    );
-  }
+                            item: resultObj.item,
+                            matches: resultObj.matches,
+                            highlighted: index === highlightedIndex,
+                            itemProps: getItemProps({
+                              item: resultObj.item,
+                              index,
+                            }),
+                          }),
+                          {key: `${resultObj.item.title}-${index}`}
+                        );
+                      })
+                    )}
+                    {!isLoading && props.resultFooter ? (
+                      <ResultFooter>{props.resultFooter}</ResultFooter>
+                    ) : null}
+                  </DropdownBox>
+                )}
+              </SearchSources>
+            ) : null}
+          </SearchWrapper>
+        );
+      }}
+    </AutoComplete>
+  );
 }
 
-export default withRouter(Search);
+const WithRouterSearch = withRouter(Search);
+export {WithRouterSearch as Search};
```

Comment:
```suggestion
export {WithRouterSearch as Search, SearchProps, DefaultSearchProps};
```

Reply:
Do we have a styleguide for this? Could be useful to define this